src/Symfony/Component/VarDumper/VarDumper.php

```diff
@@ -53,8 +47,66 @@ public static function dump($var)
     public static function setHandler(callable $callable = null)
     {
         $prevHandler = self::$handler;
+
+        // Prevent replacing the handler with expected format as soon as the env var was set:
+        if (isset($_SERVER['VAR_DUMPER_FORMAT'])) {
+            return $prevHandler;
+        }
+
         self::$handler = $callable;
 
         return $prevHandler;
     }
+
+    private static function register(): void
+    {
+        $cloner = new VarCloner();
+        $cloner->addCasters(ReflectionCaster::UNSET_CLOSURE_FILE_INFO);
+
+        switch ($_SERVER['VAR_DUMPER_FORMAT'] ?? null) {
+            case 'html':
+                $dumper = new HtmlDumper();
+                break;
+            case 'cli':
+                $dumper = new CliDumper();
+                break;
+            case 'server':
+                $host = $_SERVER['VAR_DUMPER_SERVER'] ?? '127.0.0.1:9912';
```

Comment:
can't we use some kind of DSN instead of adding a new env var?
E.g. `VAR_DUMPER_FORMAT=tcp://127.0.0.1:9912`
What about supporting `file://...?format=html` also to dump to a file?
Of course we should keep compat with the special `html` format.

Reply:
This is not entirely a new var, as it's the one we use as default when starting the server, either [within Symfony full-stack](https://github.com/symfony/symfony/blob/d12eabc9fbd0455f67221fe01022d650e753119b/src/Symfony/Bundle/DebugBundle/Resources/config/services.php#L33) or with [the binary](https://github.com/symfony/symfony/blob/e60a876201b5b306d0c81a24d9a3db997192079c/src/Symfony/Component/VarDumper/Resources/bin/var-dump-server#L50). So using this env var would be consistent.

Should we support both `VAR_DUMPER_FORMAT=server` for simplicity sake of writing it (which still fallbacks to `VAR_DUMPER_SERVER` env var is set) and `VAR_DUMPER_FORMAT=tcp://127.0.0.1:9912`?

About `file://...?format=html`, you mean directly dump into a file without using the server dumper, right?
I'm all in for that too, but in another PR 😄

Reply:
Support for `VAR_DUMPER_FORMAT=tcp://127.0.0.1:9912` format added.
`VAR_DUMPER_FORMAT=server` is kept, for the simplicity of writing it when you're already relying on `VAR_DUMPER_SERVER` env var/.env value or just using the default value.